tests/e2e/actions.go

```diff
@@ -546,17 +549,122 @@ func (tr *TestConfig) getConsumerGenesis(providerChain, consumerChain ChainID, t
 		log.Fatal(err, "\n", string(bz))
 	}
 
-	// only needed when consumer is running v3.3.x and later
-	if tr.transformGenesis {
+	if tr.transformGenesis || needsGenesisTransform(target.GetTargetConfig()) {
 		return string(tr.transformConsumerGenesis(consumerChain, bz, target))
+	} else {
+		fmt.Println("No genesis transformation performed")
 	}
 	return string(bz)
 }
 
+// needsGenesisTransform tries to identify if a genesis transformation should be performed
+func needsGenesisTransform(cfg TargetConfig) bool {
+
+	// no genesis transformation needed for same versions
+	if cfg.consumerVersion == cfg.providerVersion {
+		return false
+	}
+
+	// use v4.0.0 (after genesis transform breakages) for the checks if 'latest' is used
+	consumerVersion := cfg.consumerVersion
+	if cfg.consumerVersion == "latest" {
+		consumerVersion = "v4.0.0"
+	}
+	providerVersion := cfg.providerVersion
+	if cfg.providerVersion == "latest" {
+		providerVersion = "v4.0.0"
+	}
+
+	if !semver.IsValid(consumerVersion) || !semver.IsValid(providerVersion) {
+		fmt.Println("unable to identify the need for genesis transformation: invalid sem-version", consumerVersion, providerVersion)
+		return false
+	}
+
+	breakages := []string{"v3.0.0", "v3.3.0", "v4.0.0"}
+	for _, breakage := range breakages {
+		if semver.Compare(consumerVersion, breakage) < 0 && semver.Compare(providerVersion, breakage) >= 0 {
```

Comment:
I'd prefer to put both conditions into the same conditional to reduce code duplication in the inner part